> Hey @keith, can you clarify what the issue is here? With the ios_static_framework you added, there should be a single dep on TensorFlowLite swift_library. That library should (see note below) have a single dep on tensorflow_lite_c  cc_library. When the framework is built, it should bundle both the swift_library and cc_library without needing to specify avoid_deps. Curious what build issues you were seeing?

> Got a comment from merging this internally - what was the initial reason for bundling both TensorFlowLite_framework and TensorFlowLiteC_framework, and not just bundling TensorFlowLite_framework alone? Maybe I missed something..

So the issue is when you produce a Swift static framework, where it publicly depends on another module (it's public in this case because the Swift TF implementation exposes symbols from the `cc_library`) you must vendor that module as a separate framework since a single framework cannot expose multiple modules AFAIK

Here's a sample project demoing the issue [SampleTFProject.zip](https://github.com/tensorflow/tensorflow/files/4659505/SampleTFProject.zip)

I built the framework included in this project without the `avoid_deps` line. Here's the specific failure:

```
/tmp/SampleTFProject/TensorFlowLite.framework/Modules/TensorFlowLite.swiftmodule/x86_64.swiftinterface:6:8: error: no such module 'TensorFlowLiteC'
import TensorFlowLiteC
       ^
/tmp/SampleTFProject/SampleTFProject/AppDelegate.swift:10:8: error: failed to load module 'TensorFlowLite'
import TensorFlowLite
       ^
/tmp/SampleTFProject/TensorFlowLite.framework/Modules/TensorFlowLite.swiftmodule/x86_64.swiftinterface:6:8: error: no such module 'TensorFlowLiteC'
import TensorFlowLiteC
       ^
```

This makes sense since as far as this iOS project is concerned it's true that `TensorFlowLiteC` doesn't exist. If there is a workaround for this where we could somehow create a single framework that also bundled this module I would love to know about it and get rules_apple updated to produce it!

--- a/tensorflow/lite/experimental/swift/BUILD.apple
+++ b/tensorflow/lite/experimental/swift/BUILD.apple
@@ -2,7 +2,7 @@
 
 load("//tensorflow/lite:special_rules.bzl", "ios_visibility_whitelist", "tflite_ios_lab_runner")
 load("//tensorflow/lite/experimental/ios:ios.bzl", "TFL_DEFAULT_TAGS", "TFL_DISABLED_SANITIZER_TAGS", "TFL_MINIMUM_OS_VERSION")
-load("@build_bazel_rules_apple//apple:ios.bzl", "ios_application", "ios_unit_test")
+load("@build_bazel_rules_apple//apple:ios.bzl", "ios_application", "ios_static_framework", "ios_unit_test")
 load("@build_bazel_rules_swift//swift:swift.bzl", "swift_library")
 
 package(
@@ -36,6 +36,16 @@ swift_library(
     deps = [
         "//tensorflow/lite/experimental/delegates/coreml:coreml_delegate",
         "//tensorflow/lite/experimental/ios:tensorflow_lite_c",
+    ],
+)
+
+# bazel build -c opt --config=ios_fat //tensorflow/lite/experimental/swift:TensorFlowLite_framework
+ios_static_framework(
+    name = "TensorFlowLite_framework",
+    bundle_name = "TensorFlowLite",
+    minimum_os_version = TFL_MINIMUM_OS_VERSION,
+    deps = [
+        ":TensorFlowLite",
     ],
 )
 

--- a/tensorflow/lite/experimental/swift/BUILD.apple
+++ b/tensorflow/lite/experimental/swift/BUILD.apple
@@ -42,6 +42,9 @@ swift_library(
 # bazel build -c opt --config=ios_fat //tensorflow/lite/experimental/swift:TensorFlowLite_framework
 ios_static_framework(
     name = "TensorFlowLite_framework",
+    avoid_deps = [
+        "//tensorflow/lite/experimental/ios:tensorflow_lite_c",
+    ],
     bundle_name = "TensorFlowLite",
     minimum_os_version = TFL_MINIMUM_OS_VERSION,
     deps = [

